File: springboot/springboot-gradle/src/main/java/com/example/springbootgradle/DemoApplication.java
package com.example.springbootgradle;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

import java.util.Map;

@Controller
@SpringBootApplication
public class DemoApplication {

    @GetMapping("/")
    private String html(Map<String, Object> map) {
        map.put("key", "springboot-gradle");
        return "templates";
    }

    public static void main(String[] args) {
        SpringApplication.run(DemoApplication.class, args);
    }
}

File: springboot-kotlin/src/main/java/com/example/springbootkotlin/DemoApplication.kt
package com.example.springbootkotlin

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping

@Controller
@SpringBootApplication
open class DemoApplication {

    @GetMapping("/")
    private fun html(map: MutableMap<String, Any>): String {
        map["key"] = "springboot-kotlin"
        return "templates"
    }

    companion object {

        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(DemoApplication::class.java, *args)
        }
    }
}
